### PR TITLE
fix for overlapping partitions

### PR DESCRIPTION
**Pull request by esoroush**

### CODE REVIEW

File: oss_src/sframe/spark_unity.cpp [hunk 1]
```diff
@@ -612,11 +612,17 @@ int tordd_main(std::string & _output_directory, size_t & numPartitions, size_t &
...
   auto partition_size = sframe_len/numPartitions;
-  if(partId < (sframe_len % numPartitions)) { 
+  auto remainder = sframe_len % numPartitions;
+  size_t row_start = partition_size * partId; 
+  if(partId < remainder) { 
     partition_size +=1;
+    row_start = partition_size * partId; 
+  } else { 
+    row_start = partition_size * partId + remainder;
   }
```
Comment (ylow) [inline oss_src/sframe/spark_unity.cpp]: This whole logic is really annoying. 
What about something simple like:
```
auto partition_size = sframe_len / numPartitions;
row_start = partition_size * partId;
if (partId != numPartitions - 1) {
    row_end = row_start + partition_size;
} else {
    // last partition
    row_end = sframe_len;
}
```
Yes this has a slight degree of imbalance; but no more than numPartitions off. But the logic is much simpler and easier to show that it is correct.

**Comment by esoroush** [comment]
Hoyt, this is the fix for this issue: https://github.com/dato-code/SFrame/issues/312
whenever we merge this fix, then I can start building for linux/mac and collecting cy_spark_unity.so artifacts and generate a new spark_unity.jar.

File: oss_src/sframe/spark_unity.cpp (hunk 1 above)
Comment (hoytak) [inline oss_src/sframe/spark_unity.cpp]: Or just do: 

```
row_start = ((part_Id+1) * sframe_len) / num_partitions;
row_end = (part_id * sframe_len) / num_partitions;
```

Comment (esoroush) [inline oss_src/sframe/spark_unity.cpp]: Thanks guys for comments. I changed it to the simpler version.

**Pull request merged**